### DISCUSSION TITLE
When should I run MagickNET.Initialize();

### DISCUSSION THREAD

**Discussion by doggy8088** [dlemstra/Magick.NET#1573] [General] [answered]
I'm trying to resize an image, but I forget to initialize Magick.NET, it still works.

```cs
using var image = new MagickImage(@"G:\SNAG-0111.jpg");
image.Quality = 90;
image.Resize(1000, default(int));
await image.WriteAsync(@"G:\SNAG-0111-1000w.jpg");
```

I'd like to know when should I run `MagickNET.Initialize();` before using Magick.NET?

**Comment by dlemstra** [comment, marked as answer]
You don't need to run `MagickNET.Initialize();` but you can. You should do this if you have your own custom configuration. And when you do this should be before you execute any code the creates a `MagickImage` or `MagickImageCollection`. This should be done as early as possible.

**Reply by doggy8088** [reply]
I don't know when should I define my own custom configuration. Do you have an example?

**Reply by dlemstra** [reply]
You would normally only want that if you need to customize the `ConfigurationFiles`.

**Reply by doggy8088** [reply]
I found a docs about these `ConfigurationFiles` here: https://imagemagick.org/script/resources.php
